spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/config/AdminConfiguration.java

```diff
@@ -89,64 +96,77 @@
 	private String dataSourceUrl;
 
 	@Bean
+	@ConditionalOnMissingBean
```

Comment:
I believe the rationale is that now that an end user could create their own Admin server app by starting with `@EnableDataflowAdminServer`, they might also choose to provide their own configuration to override any of these beans. cc @joshlong